gamemodes/terrortown/entities/weapons/weapon_ttt_spawneditor.lua

```diff
@@ -0,0 +1,530 @@
+SWEP.WorldModel = "models/weapons/w_toolgun.mdl"
+
+SWEP.AutoSpawnable = false
+SWEP.NoSights = true
+
+SWEP.HoldType = "pistol"
+SWEP.LimitedStock = true
+
+SWEP.Primary.Recoil = 0
+SWEP.Primary.ClipSize = -1
+SWEP.Primary.DefaultClip = -1
+SWEP.Primary.Automatic = false
+SWEP.Primary.Delay = 0.5
+SWEP.Primary.Ammo = "none"
+
+SWEP.Secondary.Recoil = 0
+SWEP.Secondary.ClipSize = -1
+SWEP.Secondary.DefaultClip = -1
+SWEP.Secondary.Automatic = false
+SWEP.Secondary.Delay = 0.5
+
+SWEP.lastReload = 0
+
+SWEP.AllowDrop = false
+
+local previewData = {}
+
+if SERVER then
+	util.AddNetworkString("weapon_ttt_spawneditor_spawninfo_ent")
+
+	---
+	-- @ignore
+	function SWEP:Deploy()
+		-- add entity which is used for the targetID integration
+		self.entSpawnInfo = ents.Create("ttt_spawninfo_ent")
+		self.entSpawnInfo:Spawn()
+
+		timer.Simple(0, function()
+			if not IsValid(self) or not IsValid(self.entSpawnInfo) then return end
+
+			net.Start("weapon_ttt_spawneditor_spawninfo_ent")
+			net.WriteEntity(self.entSpawnInfo)
+			net.Send(self:GetOwner())
+		end)
+
+		self.BaseClass.Deploy(self)
+	end
+
+	---
+	-- @ignore
+	function SWEP:OnRemove()
+		-- using the on remove hook to store the updated spawns because it is called in the following scenarios:
+		-- * manual stop of the spawn edit process
+		-- * stop of the spawn edit process triggered by a death
+		-- * stop of the spawn edit process triggered by a new round
+		-- * stop of the spawn edit process triggered by a mapchange
+		entspawnscript.UpdateSpawnFile()
+
+		-- remove entity which is used for the targetID integration
+		if not IsValid(self.entSpawnInfo) then return end
+
+		self.entSpawnInfo:Remove()
+	end
+
+	---
+	-- @ignore
+	function SWEP:OnDrop()
+		self:Remove()
+	end
+end
+
+local maxEditDistance = 1500
+
+if CLIENT then
+	local draw = draw
+	local camStart2D = cam.Start2D
+	local camEnd2D = cam.End2D
+	local camStart3D2D = cam.Start3D2D
+	local camEnd3D2D = cam.End3D2D
+	local renderPushRenderTarget = render.PushRenderTarget
+	local renderSetColorMaterial = render.SetColorMaterial
+	local renderDrawSphere = render.DrawSphere
+	local camStart3D = cam.Start3D
+	local camEnd3D = cam.End3D
+	local Vector = Vector
+	local mathPi = math.pi
+	local mathTan = math.tan
+	local ColorAlpha = ColorAlpha
+
+	local centerX = 0.5 * ScrW()
+	local centerY = 0.5 * ScrH()
+	local sphereRadius = 10
+	local tolerance = 32 * sphereRadius
+
+	local distWalls = 7.5
+
+	local colorPreview = Color(255, 255, 255, 100)
+
+	local function IsHighlighted(pos, scPos)
+		local dist3d = LocalPlayer():EyePos():Distance(pos)
+
+		if dist3d > maxEditDistance then
+			return false, dist3d
+		end
+
+		if math.Distance(scPos.x, scPos.y, centerX, centerY) > tolerance * ScrW() / LocalPlayer():GetFOV() / dist3d then
+			return false, dist3d
+		end
+
+		return true, dist3d
+	end
+
+	local function PaintSpawns(spawnType, entTable, color, proximitySpawns)
+		local scPos
+
+		for entType, spawns in pairs(entTable) do
```

Comment:
See the note in my PR. I don't think having two parallel data systems is a good idea. And since this is only an admin feature, I think the performance penalty is alright